components/modules/file.c

```diff
@@ -116,6 +117,30 @@ static int file_fsinfo( lua_State* L )
 }
 
 
+static int file_mkdir( lua_State *L )
+{
+  const char *name = luaL_checkstring(L, 1);
+  unsigned mode = luaL_optint(L, 2, 0777);
+  int ret = mkdir(name, mode);
+  if (ret != 0)
```

Comment:
You know I'm not a firmware programmer 😉 but I'd write this like so:

```c++
if (mkdir(name, mode) != 0) {
  return luaL_error(L, "failed to create directory '%s'; code %d", name, errno);
}
return 0;
```

- you use `if`/`else` here but only `if` in `file_rmdir` (inconsistent style)
- no need for the local `ret` variable
- `{}` even for single line `if` blocks, seems consistent with the other code in `file.c`

Reply:
You keep claiming not to be one, yet the quality I've seen from you surpasses that of many a professional firmware programmer I've come across.

All fair points; I've updated accordingly, thanks.

Oh, and when you're ready, let's do the `dev-esp32` -> `dev-esp32-idf3-final` + `dev-esp32-idf4` -> `dev-esp32` renames. As far as I'm aware the idf4 branch is production ready (or at least I hope so, seeing as we've been shipping devices based off it for a while at $work 😆).

Reply:
> Oh, and when you're ready, let's do the `dev-esp32` -> `dev-esp32-idf3-final` + `dev-esp32-idf4` -> `dev-esp32` renames.

Oh absolutely, I am not opposed to that. The fact that neither https://github.com/marcelstoer/docker-nodemcu-build/issues/101 nor https://github.com/marcelstoer/nodemcu-custom-build/issues/41 are ready, doesn't mean the progress of _this_ project should be held hostage.